.circleci/config.yml

```diff
@@ -1,20 +1,61 @@
 version: 2
 
+commands:
+  restore_cache:
+    steps:
+      - restore_cache:
+          name: 'Restore cache: python env'
+          keys:
+            # These caches are persisted for 15 days. If you need to reset the cache,
+            # just change the "a" to any other value which hasn't been used recently.
+            - 'venv-a-{{ checksum "requirements.txt" }}'
+  save_cache:
+    steps:
+      - save_cache:
+          name: 'Save cache: python env'
+          key: 'venv-a-{{ checksum "requirements.txt" }}'
+          paths:
+            - .setup/venv
+
 jobs:
-  build:
+  lint:
     docker:
-      - image: circleci/python:3.7.4
+      - image: 'circleci/python:3.7'
     steps:
-        - checkout
-        - run:
-            name: Install core utilities
-            command: make install
-        - run:
-            name: Setup the Virtual Environment
-            command: make venv && source venv/bin/activate
-        - run:
-            name: Install dependencies
-            command: pipenv run -- make deps
-        - run:
-            name: CLI Tests
-            command: make ci
+      - checkout
+      - run:
+          name: Install core utilities
+          command: make install
+      - run:
+          name: Setup the Virtual Environment
+          command: make venv && source venv/bin/activate
+      - run:
+          name: Install dependencies
+          command: pipenv run -- make deps
+      - run:
+          name: Linting
+          command: pipenv run -- make lint
+  unit_tests:
+    docker:
+      - image: 'circleci/python:3.7'
+    steps:
+      - checkout
+      - run:
+          name: Install core utilities
+          command: make install
+      - run:
+          name: Setup the Virtual Environment
+          command: make venv && source venv/bin/activate
+      - run:
+          name: Install dependencies
+          command: pipenv run -- make deps
+      - run:
+          name: Linting
```

Comment:
This should have a different name.

Reply:
Yeah! The whole point of the change is to show in the 'checks' that a certain step is failing (tests vs lint). I don't know how we can make the actual job read-only though.

Reply:
Yeah, I just mean this part is named `Linting` but it's running `make test`, I think this was a bad copy/paste. It should be named `Unit tests` or something.